Extensions/HttpClientExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.Exchange.WebServices.Autodiscover;
+using Microsoft.Exchange.WebServices.Data;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Exchange.WebServices.NETStandard.Interfaces;
+using Microsoft.Exchange.WebServices.NETStandard.Core;
+
+namespace Microsoft.Exchange.WebServices.NETStandard.Extensions
+{
+    public static class HttpClientExtensions
+    {
+        internal static HttpClientHandler CreateHttpClientHandler(string httpClientName)
+        {
+            switch (httpClientName)
+            {
+                case ExchangeService.HttpClientName:
+                    {
+                        return new HttpClientHandler()
+                        {
+                            UseCookies = false,
+                            AutomaticDecompression = System.Net.DecompressionMethods.Deflate | System.Net.DecompressionMethods.GZip,

```

Comment:
Should it be same between 2 clients?
Also, auto-redirect normally is actively used for autodiscover - no error here?

Reply:
according to the old code autodiscovery service didn't set AutomaticDecompression

Reply:
What about redirect?

Reply:
in old code AllowAutoRedirect was set to false for autodiscovery service